frontend/dockerfile/dockerfile2llb/directives.go

```diff
@@ -0,0 +1,38 @@
+package dockerfile2llb
+
+import (
+	"bufio"
+	"io"
+	"regexp"
+	"strings"
+)
+
+const keySyntax = "syntax"
+
+var reDirective = regexp.MustCompile(`^#\s*([a-z][a-z0-9]*)\s*=\s*(.+?)\s*$`)
+
+func DetectSyntax(r io.Reader) (string, string, bool) {
+	directives := ParseDirectives(r)
+	if len(directives) == 0 {
+		return "", "", false
+	}
+	v, ok := directives[keySyntax]
+	if !ok {
+		return "", "", false
+	}
+	p := strings.SplitN(v, " ", 2)
+	return p[0], v, true
+}
+
+func ParseDirectives(r io.Reader) map[string]string {
+	m := map[string]string{}
+	s := bufio.NewScanner(r)
+	for s.Scan() {
+		match := reDirective.FindStringSubmatch(s.Text())
```

Comment:
Needs to handle uppercase too, so it should be `strings.ToLower(s.Text())` or `reDirective.FindSubmatch(bytes.ToLower(s.Bytes()))` (and then cast `string(m)` and `string(match[2])`)

Reply:
Or just change the regexp.

Reply:
Updated